### ISSUE TITLE
required() : Simplify the check for a non-empty string

### ISSUE THREAD

**Issue by kevinleguillou** [fabian-hiller/valibot#237]
Hello ! I'm coming from Modularforms and I encountered an issue using Valibot.
My issue comes from the fact that if you try to make a field required without any specific requirement, you would use :
```JS
const RegisterSchema = object({
  name: string()
})
```
However if you try to validate that field with an empty text input, it will not throw an error since the field is indeed a string, but an empty one. Since by default fields are required unless wrapped with optional() I was confused.
This means you need to write a required field like this :
```JS
const RegisterSchema = object({
  name: string([minLength(1, "This field is required")])
})
```
But then the validation error will read `min_length` which is not really what I want to show to the user.

For now I've created the following validation function, but it would be nice to have a standard for this.

```JS
export const required = () => (input: string) => {
  if (input === '') {
    return getPipeIssues('required', 'This field is required', input)
  }
  return getOutput(input)
}

const RegisterSchema = object({
  name: string([required()])
})
```

---

**Basically there's no way to create a standard "required" (meaning non-empty) error.**
I think it would be a good addition to the library since having a specific error key would make it easier to _translate_ and there would be no workaround with `minLength` that has less semantic value. I think the existence of `optional`, `nonOptional` and `required` functions (which is very different from Yup) made me confused too 😅 .

I'm used to Yup and its `string().required()` schema so maybe there's something similar in Valibot that I missed.

**Comment by fabian-hiller** [comment]
What is the difference between `string([required()])` and `string([minLength(1, "This field is required")])`? Isn’t it completely the same?

**Comment by kevinleguillou** [comment]
Sure `minLength(1, "Custom message")` works, but it's easier to translate with a specific error, since that would mean the ValiError would throw a unique code. Adding local hardcoded messages in English is less practical than relying on a standard key.

Right now if I translate the errors, I would translate `minLength` error as "This field must be at least X characters long" (in several languages), and make a unique edge case for catching a "required" error. It's more of a language thing.

**Comment by fabian-hiller** [comment]
Thanks for your feedback on this. There was already a very similar request in #171. I like your point about the code and the error message. I will think about it. Until then, since Valibot is modular, you can just add `required` yourself. Here is the code for it:

```ts
import { type ErrorMessage, getOutput, getPipeIssues, type PipeResult } from 'valibot';

/**
 * Creates a validation function that validates the existence of an value.
 *
 * @param error The error message.
 *
 * @returns A validation function.
 */
export function required<TInput extends string | number | bigint | any[]>(
  error?: ErrorMessage
) {
  return (input: TInput): PipeResult<TInput> =>
    (!input && input !== 0 && input !== 0n) ||
    (Array.isArray(input) && !input.length)
      ? getPipeIssues('required', error || 'Value is required', input)
      : getOutput(input);
}
```

**Comment by fabian-hiller** [comment]
`empty` and `nonEmpty` are available in the latest version.
